pkg/pod/script.go

```diff
@@ -61,7 +61,21 @@ func convertScripts(shellImage string, steps []v1alpha1.Step) (*corev1.Container
 		VolumeMounts: []corev1.VolumeMount{scriptsVolumeMount},
 	}
 
-	var containers []corev1.Container
+	convertedStepContainers := convertListOfSteps(steps, &placeScriptsInit, &placeScripts, "script")
+	sidecarContainers := convertListOfSteps(sidecars, &placeScriptsInit, &placeScripts, "sidecar-script")
+
+	if placeScripts {
+		return &placeScriptsInit, convertedStepContainers, sidecarContainers
+	}
+	return nil, convertedStepContainers, sidecarContainers
+}
+
+// convertListOfSteps does the heavy lifting for convertScripts.
+//
+// It iterates through the list of steps (or sidecars), generates the script file name and heredoc termination string,
+// adds an entry to the init container args, sets up the step container to run the script, and sets the volume mounts.
+func convertListOfSteps(steps []v1alpha1.Step, initContainer *corev1.Container, placeScripts *bool, format string) []corev1.Container {
```

Comment:
suggest "namePrefix" instead of "format" here, since the value isn't expected to be a format string.